File: ClimbLog.Api/ClimbLog.Application/Features/Routes/Commands/AddRoute/AddRouteCommand.cs
using MediatR;

namespace ClimbLog.Application.Features.Routes.Commands.AddRoute;

public class AddRouteCommand : IRequest<Unit>
{
    public required string GymName { get; set; }
    public decimal XCord { get; set; }
    public decimal YCord { get; set; }
    public int GradeId { get; set; }
    public int PointValue { get; set; }
    public int? StandardGradeId { get; set; }
    public int? CompetitionId { get; set; }
    public int? SectorId { get; set; }
}

File: climb-log-flutter/lib/climasys/widgets/upsert_route_dialog/models/add_route_command.dart
class AddRouteCommand {
  final String gymName;
  final double xCord;
  final double yCord;
  final int gradeId;
  final int? standardGradeId;
  final int? competitionId;
  final int pointValue;
  final int? sectorId;

  AddRouteCommand({
    required this.gymName,
    required this.xCord,
    required this.yCord,
    required this.gradeId,
    required this.pointValue,
    this.standardGradeId,
    this.competitionId,
    this.sectorId
  });

  Map<String, dynamic> toJson() =>
    {
      'gymName': gymName,
      'xCord': xCord,
      'yCord': yCord,
      'gradeId': gradeId,
      'pointValue': pointValue,
      'standardGradeId': standardGradeId,
      'competitionId': competitionId,
      'sectorId': sectorId
    };
}